src/Microsoft.Azure.WebJobs.ServiceBus/Triggers/BrokeredMessageToStringConverter.cs

```diff
@@ -2,54 +2,76 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Globalization;
 using System.IO;
+using System.Runtime.Serialization;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml;
 using Microsoft.Azure.WebJobs.Host.Converters;
 using Microsoft.ServiceBus.Messaging;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Triggers
 {
     internal class BrokeredMessageToStringConverter : IAsyncConverter<BrokeredMessage, string>
     {
-        public Task<string> ConvertAsync(BrokeredMessage input, CancellationToken cancellationToken)
+        public async Task<string> ConvertAsync(BrokeredMessage input, CancellationToken cancellationToken)
         {
             if (input == null)
             {
                 throw new ArgumentNullException("input");
             }
+            Stream stream = input.GetBody<Stream>();
+            if (stream == null)
+            {
+                return null;
+            }
 
-            if (input.ContentType == ContentTypes.TextPlain ||
-                input.ContentType == ContentTypes.ApplicationOctetStream ||
-                input.ContentType == ContentTypes.ApplicationJson)
+            TextReader reader = new StreamReader(stream, StrictEncodings.Utf8);
+            try
             {
-                Stream stream = input.GetBody<Stream>();
-                if (stream == null)
+                cancellationToken.ThrowIfCancellationRequested();
+                try
                 {
-                    return Task.FromResult<string>(null);
+                    return await reader.ReadToEndAsync();
+                }
+                catch (DecoderFallbackException)
+                {
+                    // we'll try again below
```

Comment:
It happens if the Reader can't deal with the bytes in the stream -- the message is: `Unable to translate bytes [99] at index 61 from specified code page to Unicode.` Because we don't own both ends of the pipeline, we need to deal with messages that are just plain strings (in which case this works) and those that are DataContract-serialized (and you need to go back to using `GetBody<string>()`).